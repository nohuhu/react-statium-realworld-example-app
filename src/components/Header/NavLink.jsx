import React from 'react';
import { NavLink as BaseNavLink } from 'react-router-dom';

const NavLink = props =>
  <BaseNavLink className={({ isActive }) => `nav-link ${isActive ? 'active' : ''}`}
    {...props} />;

export default NavLink;
