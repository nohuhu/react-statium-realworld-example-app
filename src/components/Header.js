import React from 'react';
import { bind } from 'statium';
import { Link } from 'react-router-dom';

import NavLink from './NavLink.js';

import { appName } from '../selectors/global.js';
import { getUser } from '../selectors/user.js';

import Userpic from './Userpic.js';

const LoggedOut = () => (
  <ul className="nav navbar-nav pull-xs-right">
    <li className="nav-item">
      <NavLink to="/">
        Home
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/login">
        Sign in
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/register">
        Sign up
      </NavLink>
    </li>
  </ul>
);

const LoggedIn = ({ user }) => (
  <ul className="nav navbar-nav pull-xs-right">
    <li className="nav-item">
      <NavLink to="/">
        Home
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/editor">
        <i className="ion-compose" />&nbsp;New Article
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/settings">
        <i className="ion-gear-a" />&nbsp;Settings
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to={`/@${user.username}`}>
        <Userpic src={user.image}
          className="user-pic"
          alt={user.username} />
        <span>
          &nbsp;
          {user.username}
        </span>
      </NavLink>
    </li>
  </ul>
);

export const Header = ({ user }) => (
  <nav className="navbar navbar-light">
    <div className="container">
      <Link to="/" className="navbar-brand">
        {appName.toLowerCase()}
      </Link>
      {user ? <LoggedIn user={user} /> : <LoggedOut />}
    </div>
  </nav>
);

export default bind({
  user: getUser,
})(Header);
