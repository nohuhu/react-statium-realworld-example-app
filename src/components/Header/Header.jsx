import React from 'react';
import { useStore } from 'statium';
import { Link } from 'react-router-dom';

import NavLink from './NavLink.jsx';

import { appName } from '../../misc/constants.js';

import Userpic from '../Userpic.jsx';

const LoggedOut = () => (
  <ul className="nav navbar-nav pull-xs-right">
    <li className="nav-item">
      <NavLink to="/">
        Home
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/login">
        Sign in
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/register">
        Sign up
      </NavLink>
    </li>
  </ul>
);

const LoggedIn = ({ user }) => (
  <ul className="nav navbar-nav pull-xs-right">
    <li className="nav-item">
      <NavLink to="/">
        Home
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/editor">
        <i className="ion-compose" />&nbsp;New Article
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to="/settings">
        <i className="ion-gear-a" />&nbsp;Settings
      </NavLink>
    </li>

    <li className="nav-item">
      <NavLink to={`/@${user.username}`}>
        <Userpic src={user.image}
          className="user-pic"
          alt={user.username} />
        <span>
          &nbsp;
          {user.username}
        </span>
      </NavLink>
    </li>
  </ul>
);

export const Header = () => {
  const { state: { user } } = useStore();

  return (
    <nav className="navbar navbar-light">
      <div className="container">
        <Link to="/" className="navbar-brand">
          {appName.toLowerCase()}
        </Link>
        {user ? <LoggedIn user={user} /> : <LoggedOut />}
      </div>
    </nav>
  );
}

export default Header;
