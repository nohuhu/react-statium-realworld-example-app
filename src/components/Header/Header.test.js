import React from 'react';
import Store from 'statium';
import { mount } from 'test/enzyme.js';

import Header from './Header.jsx';

describe("Header component", () => {
  it("should render when logged out", () => {
    const tree = mount(
      <Store state={{ user: null }}>
        <Header />
      </Store>
    );

    expect(tree.find('Header')).toMatchSnapshot();
  });

  it("should render when logged in", () => {
    const user = {
      username: 'foobaroo',
      image: 'https://foo.bar/baz.png',
    };

    const tree = mount(
      <Store state={{ user }}>
        <Header />
      </Store>
    );

    expect(tree.find('Header')).toMatchSnapshot();
  });
});
