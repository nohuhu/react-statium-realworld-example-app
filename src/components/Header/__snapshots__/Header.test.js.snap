// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render when logged in 1`] = `
<Header>
  <nav
    className="navbar navbar-light"
  >
    <div
      className="container"
    >
      <Link
        className="navbar-brand"
        to="/"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          conduit
        </a>
      </Link>
      <LoggedIn
        user={
          Object {
            "image": "https://foo.bar/baz.png",
            "username": "foobaroo",
          }
        }
      >
        <ul
          className="nav navbar-nav pull-xs-right"
        >
          <li
            className="nav-item"
          >
            <NavLink
              to="/"
            >
              <NavLink
                className={[Function]}
                to="/"
              >
                <Link
                  aria-current="page"
                  className="nav-link active"
                  to="/"
                >
                  <a
                    aria-current="page"
                    className="nav-link active"
                    href="/"
                    onClick={[Function]}
                  >
                    Home
                  </a>
                </Link>
              </NavLink>
            </NavLink>
          </li>
          <li
            className="nav-item"
          >
            <NavLink
              to="/editor"
            >
              <NavLink
                className={[Function]}
                to="/editor"
              >
                <Link
                  className="nav-link "
                  to="/editor"
                >
                  <a
                    className="nav-link "
                    href="/editor"
                    onClick={[Function]}
                  >
                    <i
                      className="ion-compose"
                    />
                     New Article
                  </a>
                </Link>
              </NavLink>
            </NavLink>
          </li>
          <li
            className="nav-item"
          >
            <NavLink
              to="/settings"
            >
              <NavLink
                className={[Function]}
                to="/settings"
              >
                <Link
                  className="nav-link "
                  to="/settings"
                >
                  <a
                    className="nav-link "
                    href="/settings"
                    onClick={[Function]}
                  >
                    <i
                      className="ion-gear-a"
                    />
                     Settings
                  </a>
                </Link>
              </NavLink>
            </NavLink>
          </li>
          <li
            className="nav-item"
          >
            <NavLink
              to="/@foobaroo"
            >
              <NavLink
                className={[Function]}
                to="/@foobaroo"
              >
                <Link
                  className="nav-link "
                  to="/@foobaroo"
                >
                  <a
                    className="nav-link "
                    href="/@foobaroo"
                    onClick={[Function]}
                  >
                    <Userpic
                      alt="foobaroo"
                      className="user-pic"
                      src="https://foo.bar/baz.png"
                    >
                      <img
                        alt="foobaroo"
                        className="user-pic"
                        src="https://foo.bar/baz.png"
                      />
                    </Userpic>
                    <span>
                       
                      foobaroo
                    </span>
                  </a>
                </Link>
              </NavLink>
            </NavLink>
          </li>
        </ul>
      </LoggedIn>
    </div>
  </nav>
</Header>
`;

exports[`Header component should render when logged out 1`] = `
<Header>
  <nav
    className="navbar navbar-light"
  >
    <div
      className="container"
    >
      <Link
        className="navbar-brand"
        to="/"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          conduit
        </a>
      </Link>
      <LoggedOut>
        <ul
          className="nav navbar-nav pull-xs-right"
        >
          <li
            className="nav-item"
          >
            <NavLink
              to="/"
            >
              <NavLink
                className={[Function]}
                to="/"
              >
                <Link
                  aria-current="page"
                  className="nav-link active"
                  to="/"
                >
                  <a
                    aria-current="page"
                    className="nav-link active"
                    href="/"
                    onClick={[Function]}
                  >
                    Home
                  </a>
                </Link>
              </NavLink>
            </NavLink>
          </li>
          <li
            className="nav-item"
          >
            <NavLink
              to="/login"
            >
              <NavLink
                className={[Function]}
                to="/login"
              >
                <Link
                  className="nav-link "
                  to="/login"
                >
                  <a
                    className="nav-link "
                    href="/login"
                    onClick={[Function]}
                  >
                    Sign in
                  </a>
                </Link>
              </NavLink>
            </NavLink>
          </li>
          <li
            className="nav-item"
          >
            <NavLink
              to="/register"
            >
              <NavLink
                className={[Function]}
                to="/register"
              >
                <Link
                  className="nav-link "
                  to="/register"
                >
                  <a
                    className="nav-link "
                    href="/register"
                    onClick={[Function]}
                  >
                    Sign up
                  </a>
                </Link>
              </NavLink>
            </NavLink>
          </li>
        </ul>
      </LoggedOut>
    </div>
  </nav>
</Header>
`;
